pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -57,43 +58,52 @@ func init() {
 			// buffer's not big enough
 			bufferSize += bufferIncrement
 			continue
+		} else if regerr != nil {
+			return regerr
 		}
 		break
 	}
 	clist := winutil.ConvertWindowsStringList(counterlist)
 	for i := 0; i < len(clist); i += 2 {
 		ndx, _ := strconv.Atoi(clist[i])
-		counterToIndex[clist[i+1]] = ndx
+		counterToIndex[clist[i+1]] = append(counterToIndex[clist[i+1]], ndx)
 	}
-
+	return nil
 }
 
 // GetCounterSet returns an initialized PDH counter set.
 func GetCounterSet(className string, counterName string, instanceName string, verifyfn CounterInstanceVerify) (*PdhCounterSet, error) {
 	var p PdhCounterSet
 	p.countermap = make(map[string]PDH_HCOUNTER)
-	var ndx int
 	var err error
-	if ndx = getCounterIndex(className); ndx == -1 {
-		return nil, fmt.Errorf("Class name not found: %s", className)
-	}
-	p.className, err = pdhLookupPerfNameByIndex(ndx)
+
+	// the counter index list may be > 1, but for class name, only take the first
+	// one.  If not present at all, try the english counter name
+	ndxlist, err := getCounterIndex(className)
 	if err != nil {
 		return nil, err
 	}
-	if ndx = getCounterIndex(counterName); ndx == -1 {
-		return nil, fmt.Errorf("Class name not found: %s", counterName)
-	}
-	p.counterName, err = pdhLookupPerfNameByIndex(ndx)
-	if err != nil {
-		return nil, err
+	if ndxlist == nil || len(ndxlist) == 0 {
+		log.Warnf("Didn't find counter index for %s, attempting english counter", className)
+		p.className = className
+	} else {
+		if len(ndxlist) > 1 {
+			log.Warnf("Class %s had multiple (%d) indices, using first", className, len(ndxlist))
+		}
+		ndx := ndxlist[0]
```

Comment:
When we make the counter path, we loop through, and there is a need.  You're missing something, but that's because it's really screwy.  Let me add some documentation along side the code to see if it clarifies.